src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -145,15 +144,23 @@ export class DefaultTelemetryService implements TelemetryService {
         this._eventQueue.length = 0
     }
 
-    private async flushRecords(): Promise<void> {
+    private async flushRecords(overrideEvents?: MetricDatum[]): Promise<void> {
```

Comment:
The `overrideEvents` case doesn't actually "flush" anything, it's more like a direct publish. 

It's a private method so the semantics can be a bit messy. But a brief doc might be worthwhile.
```suggestion
    /** @param overrideEvents publish these events, don't flush this._eventQueue. */
    private async flushRecords(overrideEvents?: MetricDatum[]): Promise<void> {
```

Can be improved later, but the comment avoids some confusion meanwhile.